hal/src/dmac/channel.rs

```diff
@@ -344,11 +361,52 @@ impl<Id: ChId> Channel<Id, Busy> {
     /// A `Channel` with a `Ready` status, ready to be reused by a new
     /// [`Transfer`](super::transfer::Transfer)
     #[inline]
-    pub(crate) fn free(self, dmac: &DMAC) -> Channel<Id, Ready> {
-        while !self.xfer_complete(dmac) {}
+    pub(crate) fn free(self) -> Channel<Id, Ready> {
+        while !self.xfer_complete() {}
         Channel {
             _id: self._id,
             _status: PhantomData,
         }
     }
+
+    #[inline]
+    #[cfg(any(feature = "samd11", feature = "samd21"))]
+    pub fn callback(&mut self) {
```

Comment:
Is this just a placeholder? What does it do?

Reply:
This is still unfinished, but it's intended to be called by `Transfer::callback`, which should be placed in the DMAC interrupt handler. It will take care of calling the `Transfer`'s `waker` if it exists.